src/components/WalletList.tsx

```diff
@@ -0,0 +1,65 @@
+import React from 'react';
+import './WalletList.css';
+
+type WalletEntryParams = {
+  name: string;
+  uuid: string;
+  rdns: string;
+  icon: string;
+  extensionId?: string;
+};
+
+export type WalletMapEntry = {
+  params: WalletEntryParams;
+  eventName: string;
+};
+
+type WalletListProps = {
+  wallets: Record<string, WalletMapEntry>;
+  handleClick: (extensionId?: string) => Promise<void>;
+};
+
+const WalletList: React.FC<WalletListProps> = ({ wallets, handleClick }) => {
+  const walletEntries = Object.entries(wallets);
+
+  if (!walletEntries.length) {
+    return null;
+  }
+  const onClickHandler = (extensionId?: string) => {
+    handleClick(extensionId).catch((error) => {
```

Comment:
should probably just make this handler async and then try await catch this instead (for better stack trace, but honestly it doesn't really matter)

Reply:
the `<button>` `onClick` handler will not accept a promise, because of an eslint rule apparently
<img width="775" alt="Screenshot 2024-12-13 at 10 38 06" src="https://github.com/user-attachments/assets/be704234-9d71-416d-ae02-86507c5ea79e" />
so making onClickHandler async will make compiler throw up 😢